collector/lib/CollectorConfig.cpp

```diff
@@ -278,41 +279,27 @@ void CollectorConfig::InitCollectorConfig(CollectorArgs* args) {
 }
 
 void CollectorConfig::HandleAfterglowEnvVars() {
-  if (!set_enable_afterglow) {
-    enable_afterglow_ = false;
-  }
-
-  if (const char* afterglow_period = std::getenv("ROX_AFTERGLOW_PERIOD")) {
-    afterglow_period_micros_ = static_cast<int64_t>(atof(afterglow_period) * 1000000);
-  }
+  constexpr int64_t SECOND = 1'000'000;
+  constexpr int64_t max_afterglow_period_micros = 300 * SECOND;  // 5 minutes
 
-  const int64_t max_afterglow_period_micros = 300000000;  // 5 minutes
-
-  if (afterglow_period_micros_ > max_afterglow_period_micros) {
-    CLOG(ERROR) << "User set afterglow period of " << afterglow_period_micros_ / 1000000
-                << "s is greater than the maximum allowed afterglow period of " << max_afterglow_period_micros / 1000000 << "s";
-    CLOG(ERROR) << "Setting the afterglow period to " << max_afterglow_period_micros / 1000000 << "s";
-    afterglow_period_micros_ = max_afterglow_period_micros;
-  }
-
-  if (enable_afterglow_ && afterglow_period_micros_ > 0) {
-    CLOG(INFO) << "Afterglow is enabled";
-    return;
-  }
-
-  if (!enable_afterglow_) {
-    CLOG(INFO) << "Afterglow is disabled";
-    return;
-  }
+  afterglow_period_micros_ = static_cast<uint64_t>(afterglow_period.value() * SECOND);
 
   if (afterglow_period_micros_ < 0) {
-    CLOG(ERROR) << "Invalid afterglow period " << afterglow_period_micros_ / 1000000 << ". ROX_AFTERGLOW_PERIOD must be positive.";
+    CLOG(ERROR) << "Invalid afterglow period " << afterglow_period_micros_ / SECOND << ". ROX_AFTERGLOW_PERIOD must be positive.";
+  } else if (afterglow_period_micros_ == 0) {
+    CLOG(ERROR) << "Afterglow period set to 0.";
   } else {
-    CLOG(ERROR) << "Afterglow period set to 0";
+    if (afterglow_period_micros_ > max_afterglow_period_micros) {
+      CLOG(WARNING) << "User set afterglow period of " << afterglow_period_micros_ / SECOND
+                    << "s is greater than the maximum allowed afterglow period of " << max_afterglow_period_micros / SECOND << "s";
+      CLOG(WARNING) << "Setting the afterglow period to " << max_afterglow_period_micros / SECOND << "s";
+      afterglow_period_micros_ = max_afterglow_period_micros;
+    }
+
+    enable_afterglow_ = enable_afterglow.value();
```

Comment:
The patch introduces a change in the behavior here. _Afterglow_ used to be disabled when the period is negative or 0.

Reply:
It shouldn't, I changed the value that is stored in `enable_afterglow_` to be false by default here:
https://github.com/stackrox/collector/blob/5b53bdc154dd08840fa0e29283caecd6c7798542/collector/lib/CollectorConfig.h#L113
So if the period is negative or 0, it will go unchanged and keep afterglow disabled. If the period is properly configured or left with no configuration, `enable_afterglow_` will get changed based on the values in the EnvVars here: https://github.com/stackrox/collector/blob/5b53bdc154dd08840fa0e29283caecd6c7798542/collector/lib/CollectorConfig.cpp#L39-L40

Reply:
Bad reading of the number of spaces from my part. I thought that the following part was always executed:

```c
enable_afterglow_ = enable_afterglow.value();
```